js/utils/Translator.js

```diff
@@ -0,0 +1,393 @@
+(function (root, factory) {
```

Comment:
We should add a copyright notice at the top where we keep attribution to the original authors.